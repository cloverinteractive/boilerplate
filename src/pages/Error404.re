[@react.component]
let make = () =>
  <div className="container is-fluid">
    <h1 className="title"> "Page not found"->React.string </h1>
    <div className="content">
      <p> "The page you we're looking for does not exist."->React.string </p>
    </div>
    <BsReactHelmet defaultTitle="Oops! Page not found" />
  </div>;

let default = make;