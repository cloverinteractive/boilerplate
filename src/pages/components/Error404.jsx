import React from 'react';
import { Container, Header } from 'semantic-ui-react';
import { Helmet } from 'react-helmet';

const Error404 = () => (
  <Container text>
    <Helmet>
      <title>Oops! Page not found</title>
    </Helmet>

    <Header as="h1">Page not found</Header>
    <p>The page you we&apos;re looking for does not exist.</p>
  </Container>
);

export default Error404;
