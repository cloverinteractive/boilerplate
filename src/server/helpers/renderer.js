import { renderToString } from 'react-dom/server';
import { Helmet } from 'react-helmet';
import serialize from 'serialize-javascript';
import { isDevelopment } from 'server/helpers/env-access';

export default (Component, store) => {
  const content = renderToString(Component);
  const helmet = Helmet.renderStatic();

  const productionStyles =
    `<link rel="stylesheet" href="/vendors.css">
    <link rel="stylesheet" href="/styles.css">`;

  const styles = isDevelopment ? '' : productionStyles;

  const template =
    `<!doctype html>
    <html lang="en-US">
      <head>
        <meta charset="utf-8">
        ${helmet.link.toString()}
        ${helmet.title.toString()}
        ${helmet.meta.toString()}
        ${styles}
        <script src="/vendors.js"></script>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.__INITIAL_STATE__ = ${serialize(store.getState())};
        </script>
        <script async src="/bundle.js"></script>
      </body>
    </html>`;

  return template;
};
