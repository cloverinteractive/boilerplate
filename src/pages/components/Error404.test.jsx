import * as React from 'react';
import renderer from 'react-test-renderer';
import Error404 from './Error404';

describe('<Error404 />', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(<Error404 />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
