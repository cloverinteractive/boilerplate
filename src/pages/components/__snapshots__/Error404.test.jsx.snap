// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error404 /> it renders correctly 1`] = `
<div
  className="container is-fluid"
>
  <h1
    className="title"
  >
    Page not found
  </h1>
  <div
    className="content"
  >
    <p>
      The page you we're looking for does not exist.
    </p>
  </div>
</div>
`;
